src/main/java/uk/gov/companieshouse/ocr/api/urlvalidator/WhiteListedUrlValidator.java

```diff
@@ -0,0 +1,55 @@
+package uk.gov.companieshouse.ocr.api.urlvalidator;
+
+import java.util.Set;
+import org.apache.commons.validator.routines.UrlValidator;
+import uk.gov.companieshouse.ocr.api.OcrApiApplication;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+import java.net.MalformedURLException;
+import java.net.URL;
+
+
+public class WhiteListedUrlValidator {
```

Comment:
Is this the best name?
If it's intended for reuse, it does not need to be confined to whitelisting. Could be used for blacklisting, for example.
I'm also not sure about the UrlValidator bit of the name. It doesn't really validate the whole URL. It just validates the hostname part.

Reply:
I'll have a rethink of the names and get back to you

Reply:
UrlHostValidator?
HostNameValidator?
UrlHostChecker?